src/analysis/souffle/sensitivity_analysis.dl

```diff
@@ -0,0 +1,132 @@
+#ifndef SENSITIVITY_ANALYSIS_DL_
+#define SENSITIVITY_ANALYSIS_DL_
+//this file defines the calculations for sensitivity analysis 
+//to be used in dp calculations
+
+#include "src/analysis/souffle/taint.dl"
+
+//one isSqlOperator fact per each sql operator, currently supporting one output
+//referring to results as 'value' (sqlValue declared below) as per the discussion in the doc 
+.decl isSqlOperator(sqlOperator: Operation)
+isSqlOperator([owner, operator, result, input_list, attribute_list]) :- isOperation([owner, operator, result, input_list, attribute_list]), match("sql.*", operator). 
+
+
+.decl isSqlValue(value: AccessPath)
+isSqlValue(value) :- isSqlOperator([_, _, value, _, _]).
+
+//connecting each value to its input list 
+.decl sqlValueHasSqlInputList(value: AccessPath, sql_input: OperandList)
+sqlValueHasSqlInputList(value, input_list) :- isSqlOperator([_, _, value, input_list, _]). 
+
+//keeping track of the suffix of each input list
+//used in finding the maximm sensitvity 
+.decl sqlValueHasSqlInputListSuffix(value: AccessPath, partialList: OperandList)
+
+sqlValueHasSqlInputListSuffix(value, input_list) :-
+  sqlValueHasSqlInputList(value, input_list).
+sqlValueHasSqlInputListSuffix(value, tail) :-
+  sqlValueHasSqlInputListSuffix(value, [_, tail]).
+
+//calculate the length of each sql operator input list by iterating starting from the end
+.decl sqlOperatorInputListLength(input_list: OperandList, list_length: number)
```

Comment:
I think this is duplicated work. You can probably just use the relation already populated [here](https://github.com/google-research/raksha/blob/main/src/analysis/souffle/operations.dl#L89).

Reply:
Yes. I reused that code in the more recent commits. Thank you!